File: mu_wang/9_13/Partition Array by Odd and Even.py
class Solution:
    # @param nums: a list of integers
    # @return: nothing
    def partitionArray(self, nums):
        # write your code here
        i = 0  #pointer to the front 
        j = len(nums)-1  #pointer to the end
        while i<=j:
            if nums[i]%2==0 and nums[j] % 2==1:
                nums[i],nums[j] = nums[j],nums[i]
            elif nums[i]%2!=0:
                i+=1
            elif nums[j]%2!=1:
                j-=1

File: ziteng/Partition Array by Odd and Even.java
public class Solution {
    /**
     * @param nums: an array of integers
     * @return: nothing
     */
    public void partitionArray(int[] nums) {
        if(nums.length < 2) {
          //  nums = nums;
        } else {
            int p = 0;// odd pointer
            int q = 0;// even pointer
            if(nums[0]%2==0){
                p = 1;
                q = 0;
            } else {
                p = 0;
                q = 1;
            }
            
            while(p < nums.length) {
                if(nums[p]%2!=0 && p>q) {
                    nums[q] = nums[p] + nums[q];
                    nums[p] = nums[q] - nums[q];
                    nums[q] = nums[q] - nums[q];
                    q++;
                }
                p++;
            }
        }
        //return nums;
    }
}
